src/main/java/org/apache/jmeter/protocol/mqtt/paho/clients/BaseClient.java

```diff
@@ -21,22 +21,23 @@
 import org.eclipse.paho.client.mqttv3.MqttException;
 
 import java.io.Closeable;
-import java.util.concurrent.ConcurrentLinkedQueue;
+//import java.util.concurrent.ConcurrentLinkedQueue; ---- Root cause of high CPU utilization even for a single user run
+import java.util.concurrent.ArrayBlockingQueue; // Used to overcome the issue caused by ConcurrentLinkedQueue
 import java.util.concurrent.atomic.AtomicLong;
 
 /**
  * This is the template class for both Async and Sync MQTT clients.
  */
 public abstract class BaseClient implements MqttCallback, Closeable {
-    protected ConcurrentLinkedQueue<Message> mqttMessageStorage = null;
+    protected ArrayBlockingQueue<Message> mqttMessageStorage = new ArrayBlockingQueue<Message>(1024);
```

Comment:
Theorteically, yes. However, I have not been able to simulate a scenario where there are 1024 Messages received a single user at a single instance using JMeter.